crates/core/src/util.rs

```diff
@@ -256,29 +255,33 @@ impl TimeSource for SystemTime {
     }
 }
 
+use std::cell::RefCell;
+use std::rc::Rc;
+
 #[cfg(test)]
 #[derive(Clone)]
 pub struct MockTimeSource {
-    current_instant: Instant,
+    current_instant: Rc<RefCell<Instant>>,
 }
 
 #[cfg(test)]
 impl MockTimeSource {
     pub fn new(start_instant: Instant) -> Self {
         MockTimeSource {
-            current_instant: start_instant,
+            current_instant: Rc::new(RefCell::new(start_instant)),
         }
     }
 
-    pub fn advance_time(&mut self, duration: time::Duration) {
-        self.current_instant += duration;
+    pub fn advance_time(&self, duration: Duration) {
```

Comment:
why do you need a &self here, a &mut should do since you are not doing anything between threads

Reply:
Look at `test_cleanup` test in `received_packet_tracker`, I did the stuff with Rc<RefCell> to get that test working. Not sure of best approach.